### PR TITLE
Small fixes to static analyzer bugs

### PR DESCRIPTION
**Pull request by qarmin**
Part 1 #27638 - 3,4,6,9,10
Part 2 #27649 - 1,3,8,10,11,13
Part 3 #27668 - 4,15,17

### CODE REVIEW

File: platform/x11/power_x11.cpp [hunk 1]
```diff
@@ -115,7 +115,7 @@ bool PowerX11::make_proc_acpi_key_val(char **_ptr, char **_key, char **_val) {
 
 	*(ptr++) = '\0'; /* terminate the key. */
 
-	while ((*ptr == ' ') && (*ptr != '\0')) {
+	while ((*ptr != ' ') && (*ptr != '\0')) {
```
Comment (ShyRed) [inline platform/x11/power_x11.cpp]: Shouldn't the loop run until it hits a non-space character? 🤔 
In the following lines there is an explicit check for `'\0'` so the additional check in the loop can be removed, I think:
https://github.com/godotengine/godot/blob/5ce3999a77dd94f099c49ab841364d6cdf7c630b/platform/x11/power_x11.cpp#L122-L124
So the fix might look like
```suggestion
	while (*ptr == ' ') {
```

Comment (qarmin) [reply platform/x11/power_x11.cpp]: Right, I wrongly understand comment.

File: modules/gdscript/gdscript_editor.cpp [hunk 2]
```diff
@@ -2176,12 +2175,10 @@ static void _find_call_arguments(const GDScriptCompletionContext &p_context, con
 						return;
 					}
 				}
-				if (!_static) {
-					for (int i = 0; i < base_type.class_type->functions.size(); i++) {
-						if (base_type.class_type->functions[i]->name == p_method) {
-							r_arghint = _make_arguments_hint(base_type.class_type->functions[i], p_argidx);
-							return;
-						}
+				for (int i = 0; i < base_type.class_type->functions.size(); i++) {
```
Comment (akien-mga) [inline modules/gdscript/gdscript_editor.cpp]: CC @vnen @bojidar-bg

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
@qarmin Can I let you mark the relevant entries as fixed in the linked issues? If you want you can use the `- [ ]` syntax to create checkbox, and them tick them off (of write `- [x]` directly for the fixed ones).

**Pull request merged**